.bit/responses/3.8-Week 3 Step 8.md

```diff
@@ -50,6 +50,7 @@ const querySpec = {
 </details>
 
 ## 2: Selecting a Random Item
+> ⚠️ **Warning:** Keep in mind that *querying for ALL data* at the same time ***is not good practice.*** Data should only be queried on-demand when the user requires it. In this case, the user only needs *one random secret*, so in a "real-world application," we would utilize [built-in functionalities](https://www.mongodb.com/docs/manual/reference/operator/aggregation/sample/) for SQL or [stored procedures](https://docs.microsoft.com/en-us/azure/cosmos-db/sql/how-to-write-stored-procedures-triggers-udfs?tabs=javascript#stored-procedures). 
```

Comment:
For the built In functionality I would distinguish the different APIs. The "built-in functionalities" currently points to the Mongo DB API, which is not used in the curriculum. I would therefore put in SQL API too 

Proposal: 
In this case, the user only needs *one random secret*, so in a "real-world application," we would utilize built-in functionalities for [MongoAPI](https://www.mongodb.com/docs/manual/reference/operator/aggregation/sample/) or [stored procedures](https://docs.microsoft.com/en-us/azure/cosmos-db/sql/how-to-write-stored-procedures-triggers-udfs?tabs=javascript#stored-procedures). The situation is a bit more tricky for SQL API, you find more information in this [blog post](http://www.titov.net/2005/09/21/do-not-use-order-by-rand-or-how-to-get-random-rows-from-table/)